### DISCUSSION TITLE
Different PDF handling on SEB 3.4

### DISCUSSION THREAD

**Discussion by juliussaputra** [SafeExamBrowser/seb-win-refactoring#487] [Q&A] [answered]
Dear SEB Team

we are currently testing the new version of SEB (3.4 for Windows and 3.1.1 for macOS). We noticed the different approach of PDF Handling in SEB for windows.
On the previous version when we open a download link for PDF (for example from Moodle), SEB would display the PDF directly with internal PDF Reader and now it is blocked and shows a pop up "Download is not available on the config", but when we allow the download setting as screenshot below:
![image](https://user-images.githubusercontent.com/17081818/195552924-c66041d3-53c4-47f8-adc6-23b1a111a6b4.png)
SEB will download the file but not showing it in the browser window.

This different handling only happens on **SEB 3.4** for Windows, on MAC it works just like the previous version: PDF will be opened directly and display in SEB window.

Is there any way to get the same handling on the new version?
we are probably going for the previous for the PDF handling this semester and hope that you won't delete the previous bundle until next year :) 

Many thanks in advance
Julius

**Comment by danschlet** [comment, marked as answer]
I fixed some issues with downloads and displaying of PDF files in SEB 3.1.1 for macOS (and some in 3.1, also in the iOS version). The policy in SEB 3.1.1 for PDF files is now as follows:
- If "Allow downloading and uploading files" is disabled and "Download and open PDF files instead of displaying them inline" is disabled (this setting is currently not very useful on macOS, as it's not easily possible to permit other applications running securely during an exam), then PDF files are always displayed by SEB in the internal PDF viewer. In the iOS version this applies even if "Allow downloading and uploading files" is enabled, as the iOS version of SEB currently doesn't support downloading files (besides website generated data).
- If "Allow downloading and uploading files" is enabled, then SEB for macOS would download PDF files (and NOT display them), if the website explicitly links those files for download (when a "download" attribute or "Content-Disposition" header with attachment filename is used by the website).

**Comment by brunobaudry** [comment]
Adding comment for the Moodlers that will stumble upon this thread. To allow the PDF to be rendered by the browsers (win or mac) as described by @danschlet above. You need to make sure there are no "Force download of files" ticked (by default it is ticket for folders) in the moodle settings'.

**Reply by ghost** [reply]
> Adding comment for the Moodlers that will stumble upon this thread. To allow the PDF to be rendered by the browsers (win or mac) as described by @danschlet above. You need to make sure there are no "Force download of files" ticked (by default it is ticket for folders) in the moodle settings'.

We are using "Allow downloading and uploading files" (enabled) + "Download and open PDF files instead of displaying them inline" (disabled) + Moodle folder activity with setting "Force download files" (disabled).

My collegues tell me that at last mock exam they still saw some students with the small PDF in window.
All students are using SEB 3.4 for MacOS.

**Reply by danschlet** [reply]
Also see this issue https://github.com/SafeExamBrowser/seb-mac/issues/354 for the issue with small PDFs in the Moodle file upload activity and some versions of macOS Ventura.
